### PR TITLE
Add KafkaProducerPoolConfig

### PR DESCRIPTION
**Pull request by z0isch**
https://app.asana.com/0/1201325186562318/1204292061062064/f

We will probably want the ability to set poolSize as env variable, so this exposes a way to do that.

### CODE REVIEW

File: library/Freckle/App/Kafka.hs [hunk 1]
```diff
@@ -40,37 +43,58 @@ readKafkaBrokerAddresses t = case NE.nonEmpty $ T.splitOn "," $ T.pack t of
...
   -- ^ Maximum number of resources to keep open per stripe.  The
   -- smallest acceptable value is 1.
   --
   -- Requests for resources will block if this limit is reached on a
   -- single stripe, even if other stripes have idle resources
   -- available.
+  }
+  deriving stock (Show)
+
+-- Same defaults as `Database.Persist.Sql.ConnectionPoolConfig`
```
Comment (pbrisbin) [inline library/Freckle/App/Kafka.hs:67]:
```suggestion
-- | Same defaults as 'Database.Persist.Sql.ConnectionPoolConfig'
```

File: library/Freckle/App/Kafka.hs [hunk 2]
```diff
@@ -40,37 +43,58 @@ readKafkaBrokerAddresses t = case NE.nonEmpty $ T.splitOn "," $ T.pack t of
...
+defaultKafkaProducerPoolConfig :: KafkaProducerPoolConfig
+defaultKafkaProducerPoolConfig = KafkaProducerPoolConfig 1 600 10
+
+envKafkaProducerPoolConfig
+  :: Env.Parser Env.Error KafkaProducerPoolConfig
+envKafkaProducerPoolConfig = do
+  poolSize <- Env.var Env.auto "KAFKA_PRODUCER_POOL_SIZE" $ Env.def 10
+  pure $ defaultKafkaProducerPoolConfig {kafkaProducerPoolConfigSize = poolSize}
```
Comment (pbrisbin) [inline library/Freckle/App/Kafka.hs:75]: Whoa, we got some `ApplicativeDo` 'round here?

**Review by pbrisbin — APPROVED** [review]
LGTM.

Another speed-up tip: just leave this PR open and use a `git` extra-dep in backend. Only bump/merge/release when you're happily in prod.

File: library/Freckle/App/Kafka.hs (hunk 2 above)
Comment (z0isch) [reply library/Freckle/App/Kafka.hs:75]: `Parser` is only `Applicative`, so in order to write monadic looking parsers you have to enable it. I think we have it in some other `Parser` in `core` too.

**Pull request merged**